mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -289,7 +289,10 @@ def SYCLConstructorOp : SYCL_Op<"constructor",
     /// otherwise.
     void getEffects(::llvm::SmallVectorImpl<::mlir::SideEffects::EffectInstance<
                         ::mlir::MemoryEffects::Effect>> &effects);
-  }];      
+
+    /// Returns the index of the accessor operand.
+    unsigned getMemRefOperandIndex() const { return 0; }
```

Comment:
I'm not sure this is the ideal name for this function, as we might have a constructor receiving more than one memref.

Reply:
Changed to:
```
    /// Returns the index of the SYCL object being created.
    unsigned getOutputOperandIndex() const { return 0; }